TypeCobol.Analysis/Graph/CfgDotFileForNodeGenerator.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using TypeCobol.Analysis.Cfg;
+using TypeCobol.Compiler.CodeElements;
+using TypeCobol.Compiler.Nodes;
+
+namespace TypeCobol.Analysis.Graph
+{
+    /// <summary>
+    /// Graphviz Dot file generator for a Control Flow Graph
+    /// </summary>
+    /// <typeparam name="D"></typeparam>
+    public class CfgDotFileForNodeGenerator<D> : CfgDotFileGenerator<Node, D>
+    {
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="cfg">The underlying Control Flow Graph</param>
+        public CfgDotFileForNodeGenerator(ControlFlowGraph<Node, D> cfg) : base (cfg)
+        {
+        }
+
+        /// <summary>
+        /// Set whether full instruction must be generated are not.
+        /// If not only the instruction name will be generated.
+        /// </summary>
+        public bool FullInstruction
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// Get the string representing an instruction.
+        /// </summary>
+        /// <param name="instruction">The instruction to get the string representation.</param>
+        protected override string InstructionToString(Node instruction)
```

Comment:
Duplicate with the method in BasicBlock itself. See how we can factorize this.

Reply:
This is a overloading for Node instance which uses CodeElement. And thus this is a specialization.
BasicBlock is Generic.